list/list_resource.go

```diff
@@ -113,6 +117,9 @@ type ListResultsStream struct {
 	// To indicate a fatal processing error, push a [ListResult] that contains
 	// a [diag.ErrorDiagnostic].
 	Results iter.Seq[ListResult]
+
+	// Diagnostics report errors or warnings related to the list operation.
```

Comment:
Are these like global errors? When would a provider developer want to use this field vs. the diagnostics in `ListResult`

Reply:
Good catch – this is obsolete & removed in an upcoming commit.